operator/pkg/ciliumendpointslice/ceptocesmap.go

```diff
@@ -6,23 +6,46 @@ package ciliumendpointslice
 import (
 	capi_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
 	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
 	"github.com/cilium/cilium/pkg/lock"
 )
 
 type CEPName resource.Key
 type CESKey resource.Key
 type CESName string
+type NodeName string
+type EncryptionKey int
+type CID string
+
+// To maintain compatibility with duplicate CIDs.
+type SecIDs struct {
+	ids  map[CID]struct{}
+	ceps map[CEPName]struct{}
+}
 
 // CESToCEPMapping is used to map Cilium Endpoints to CiliumEndpointSlices and
 // retrieving all the Cilium Endpoints mapped to the given CiliumEndpointSlice.
 // This map is protected by lock for consistent and concurrent access.
 type CESToCEPMapping struct {
 	mutex lock.RWMutex
+
 	// cepNameToCESName is used to map CiliumEndpoint name to CiliumEndpointSlice name.
 	cepNameToCESName map[CEPName]CESName
 	// cesNameToCEPNameSet is used to map CiliumEndpointSlice name to all CiliumEndpoints names it contains.
 	cesNameToCEPNameSet map[CESName]map[CEPName]struct{}
 	cesData             map[CESName]CESData
+
+	// nodeNameToCEPNameSet is used to map Node name to all CiliumEndpoints on the node
+	nodeNameToCEPNameSet    map[NodeName]map[CEPName]struct{}
```

Comment:
How about grouping the information keyed by the same type into a struct? That would make the access pattern easier to reason about. For instance, something like (names TBD):

```go
type NodeInfo struct {
	CEPs   sets.Set[CEPName]
	EncKey EncryptionKey
}

type CESToCEPMapping struct {
    ...
    type CESToCEPMapping struct {
    ...
}
```